src/python/pants/engine/internals/parser.py

```diff
@@ -286,12 +287,21 @@ def error_on_imports(build_file_content: str, filepath: str) -> None:
         lineno, _ = token[2]
         if token_str != "import":
             continue
-        raise ParseError(
-            f"Import used in {filepath} at line {lineno}. Import statements are banned in "
-            "BUILD files because they can easily break Pants caching and lead to stale results. "
-            f"\n\nInstead, consider writing a macro ({doc_url('macros')}) or "
-            f"writing a plugin ({doc_url('plugins-overview')}."
-        )
+        filetype = "macro" if os.path.basename(filepath) == "macros.py" else "build"
```

Comment:
https://www.pantsbuild.org/docs/macros suggests that file must be named `macros.py` so I reckon it's safe to hardcode it here? Can't find any usages of `macros.py` in the repo. :/